src/processors/default/messagesProcessor.ts

```diff
@@ -0,0 +1,39 @@
+import { hrtime } from 'node:process';
+
+import { Table } from '../../db/db';
+import { newMessages } from '../../triggers/messages';
+import { CrdtMessage, ProcessedMessage } from '../../types/message';
+import { Clients, Processor } from '../../types/processor';
+
+
+const NAME = 'messageProcessor'
+
+export const crdtMessageProcessor: Processor = {
+
+  name: NAME,
+  trigger: newMessages,
+  // recieves messages in order of timestamp
+  // messages older than what has already been processed should not be included
+  processorFunction: async (crdtMessages: CrdtMessage[] , clients: Clients) => {
+    const processedMessages: ProcessedMessage[] = []
+
+    // process messages
+    for (const message of crdtMessages){
```

Comment:
Need to optimise to perform all updates for a single table at once, rather than one at a time